### PR TITLE
feat: add concurrency_limit option to rust capture

### PR DESCRIPTION
**Pull request by frankh**
## Problem
large numbers of concurrent requests can cause ram spikes and OOMs

allow us to set concurrency limits to control this

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: rust/capture/src/config.rs [hunk 1]
```diff
@@ -58,6 +58,8 @@ pub struct Config {
 
     #[envconfig(default = "events")]
     pub capture_mode: CaptureMode,
+
+    pub concurrency_limit: Option<usize>,
```
Comment (oliverb123) [inline rust/capture/src/config.rs:62]: I think you're better off decorating this with `envconfig` and setting a (high) default

**Review by oliverb123 — APPROVED** [review]
One comment, otherwise lgtm

File: rust/capture/src/config.rs (hunk 1 above)
Comment (frankh) [reply rust/capture/src/config.rs:62]: i did that originally but then got worried about messing up events unnecessarily - this way i can be very confident nothing breaks if we don't change that env var

what exactly does the envconfig decorator do that would help here?

Comment (frankh) [reply rust/capture/src/config.rs:62]: gonna leave it optional for now but happy to change if we want this on events too

**Pull request merged**